### ISSUE TITLE
[HOLD for payment 2024-10-24] [HOLD for payment 2024-10-16] Add LHN debugging to Debug Mode

### ISSUE THREAD

**Issue by puneetlath** [Expensify/App#46992]
We are adding a debug mode to the App here: https://github.com/Expensify/App/pull/45481

What I would really love from the Debug Mode is something that tells me why a given report is in the LHN. So if I'm seeing a report in the LHN, why is it there? And similarly if it has an RBR/GBR, what is causing that to be there?
This might be more useful in Focus Mode than in Most Recent, but I think it would help with a lot of the recent issues we've been troubleshooting.

cc @pac-guerreiro 

<details><summary>Issue Owner</summary>Current Issue Owner: @puneetlath</details>

**Comment by pac-guerreiro** [comment]
Hi! I’m Pedro Guerreiro from Callstack - expert contributor group. I’d like to work on this task!

**Comment by puneetlath** [comment]
Still waiting for the first PR to get merged.

**Comment by DylanDylann** [comment]
yeah, I will try to complete check list today

**Comment by puneetlath** [comment]
Still waiting for the first PR to get merged.

**Comment by blimpich** [comment]
@puneetlath can you respond to [this comment](https://github.com/Expensify/App/pull/48442#issuecomment-2374654016) on the PR?

**Comment by fabioh8010** [comment]
**Update:** PR is being reviewed by C+, [should be completed before next week](https://github.com/Expensify/App/pull/48442#issuecomment-2377328948).

**Comment by blimpich** [comment]
Assigning @DylanDylann since they are reviewing the PR

**Comment by blimpich** [comment]
I'll be OOO starting this afternoon until October 14th. @puneetlath can you give the final approval once the PR is ready?

**Comment by neil-marcellini** [comment]
Taking over as Ben asked and because Puneet is out as well

**Comment by puneetlath** [comment]
Thanks @neil-marcellini!

Something I realized with this, will we be able to see why a report is in the LHN without clicking into it? For example, I want to know why these reports are in my LHN, but I don't want to click into them and have OpenReport be called. 

<img width="374" alt="image" src="https://github.com/user-attachments/assets/e0288eb1-ffb1-4fd0-ab7b-59c1155c4b9a">

**Comment by DylanDylann** [comment]
Agreed. I see that we can't know why these reports display in LHN now because if we click on the report and check the debug tab on the report detail page, It means that we are focusing on this report and the reason will always be "Is temporarily focused"

**Comment by puneetlath** [comment]
@pac-guerreiro what would you think of being able to open the debug pane for a given report via right click, without clicking into it. 

<img width="387" alt="image" src="https://github.com/user-attachments/assets/e9913847-9d7c-4813-8625-be2937e209a7">

**Comment by pac-guerreiro** [comment]
@puneetlath I will give it a try to see if it won't trigger OpenReport too

**Comment by pac-guerreiro** [comment]
@puneetlath 

Sorry for the delay, I got caught with other tasks related to debug mode 😅 

Here is a preview of the change you requested:
https://github.com/user-attachments/assets/8f47255b-9d89-46d7-9270-dbbcb364a6b8

It's safe to say that opening debug page from context menu won't trigger the `OpenReport` API call

**Comment by puneetlath** [comment]
Looks great to me!

**Comment by pac-guerreiro** [comment]
@puneetlath

Awesome 😄 Should there be a new issue for this or do you want me to just add this in a new PR?

**Comment by puneetlath** [comment]
I think we can just do another PR attached to this issue. @DylanDylann can review again since they have context.

**Comment by pac-guerreiro** [comment]
@puneetlath @DylanDylann 

Here is the PR that adds the debug option to LHN context menu - https://github.com/Expensify/App/pull/50519 😄

**Comment by DylanDylann** [comment]
@pac-guerreiro Currently, we don't display the reason why RBR is displayed. But let's see logic to display RBR on LHN

https://github.com/Expensify/App/blob/12e41b177cc0a3dc1c4b26beceb28d337e69a37d/src/libs/SidebarUtils.ts#L237-L255

We have three cases where the RBR is displayed on LHN
- shouldDisplayTransactionThreadViolations return true
- hasErrors (errors come from report or reportActions. Usually, this will be returned from BE)
- hasViolations

We need to design three reason messages for each case

**Comment by puneetlath** [comment]
Yes, let's add that. I think it would be very helpful for diagnosing why RBRs are showing.

**Comment by pac-guerreiro** [comment]
@puneetlath can you create a new issue for me to implement this? 😄

**Comment by pac-guerreiro** [comment]
Thanks @puneetlath, that was fast 😄

**Comment by puneetlath** [comment]
@DylanDylann are we all done here?

**Comment by DylanDylann** [comment]
@puneetlath Do we require the regression test on debug feature issues?

**Comment by puneetlath** [comment]
Yes please. We rely on these features quite a bit, so it'd be good to have some regression testing to make sure we don't break them accidentally. Otherwise we won't realize they are broken until we need them to help us debug something.

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal

0. Enable the Debug feature
1. On the home page, open a chat report from LHN
2. Click on the header of the chat you just opened, then click on Debug
3. On the Details tab, confirm that "Visible in LHN" provides the correct reason for the report to be shown on LHN
4. Confirm that "GBR" shows the correct reason, if the report has GBR in LHN
5. Confirm that "RBR" shows the correct reason, if the report has RBR in LHN



Do we agree 👍 or 👎

**Comment by puneetlath** [comment]
@DylanDylann were there multiple PRs for this one?

**Comment by DylanDylann** [comment]
@puneetlath Initially we implemented [this first PR](https://github.com/Expensify/App/pull/48442), after that we have a new requirement to add debug option to context menu on LHN, the [second PR](https://github.com/Expensify/App/pull/50519) was created to address this. Finally, we created the [third PR](https://github.com/Expensify/App/pull/50831)  to add RBR reason to debug mode that we missed in the first PR (Note that: we already have [an issue](https://github.com/Expensify/App/issues/50665) for this problem)

**Comment by DylanDylann** [comment]
I also updated [the regression test](https://github.com/Expensify/App/issues/46992#issuecomment-2421661050) to cover all implementations

**Comment by puneetlath** [comment]
@DylanDylann I'm going to have each of the debug mode PRs be a milestone on one Upwork job. Just to make things easier for me to pay. 

The offer for is here: https://www.upwork.com/nx/wm/offer/new/1039087488. I've added 3 PRs as milestones so far. I'll add the others as milestones as we go. 

Please ping me here when you've accepted.

**Comment by DylanDylann** [comment]
@puneetlath Accepted 🙏

**Comment by puneetlath** [comment]
Paid these milestones: 
- Add LHN debugging to Debug Mode
- Add debug option to LHN context menu

Regression test issue here: https://github.com/Expensify/Expensify/issues/442178

Closing this one out. Thanks y'all!
